resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,167 @@
+
+        if is_background(sample):
+            continue
+
+        macs_list = sample.get_macs()
+        if not macs_list:
+            raise ValueError(
+                "Sample {} has no `macs14` data object!".format(sample)
+            )
+        elif len(macs_list) != 1:
+            raise ValueError(
+                "Sample {} has more than one `macs14` data objects!".format(sample)
+            )
+
+        macs14.append(macs_list[0].id)
+
+        background = sample.get_background(fail_silently=True)
+        if background:
+            if background not in samples:
+                raise ValueError(
+                    "Background of the sample {0} cannot be found in the resource you provided: "
+                    "{1}!".format(sample, resource)
+                )
+
+            relations.append(
+                ':'.join([sample.name, background.name])
+            )
+
+        collection_ids.add(get_resource_collection(sample))
+
+    auto_name, collection = get_name_collection(collection_ids, resolwe)
+
+    inputs = {
+        'reads': reads,
+        'macs14': macs14,
+        'relations': relations,
+        'name': name or auto_name,
+    }
+    geo = resolwe.get_or_run(slug='prepare-geo-chipseq', input=inputs)
+
+    if collection:
+        collection.add_data(geo)
+
+    return geo
+
+
+def prepare_geo_rnaseq(resource, name=None):
+    """Run ``Prepare GEO - RNA-Seq`` process on the resource.
+
+    This method can be used to run ``Prepare GEO - RNA-Seq`` process
+    on a single collection or a list of samples.
+
+    :param resource: resource on which prepare_geo_rnaseq will be run
+    :param str name: name of the prepare GEO tarball and table
+
+    """
+
+    reads = []
+    expressions = []
+
+    samples = get_samples(resource)
+    resolwe = get_resolwe(*samples)
+    collection_ids = set()
+
+    for sample in samples:
+        reads.append(sample.get_reads().id)
+        expressions.append(sample.get_expression().id)
+        collection_ids.add(get_resource_collection(sample))
+
+    auto_name, collection = get_name_collection(collection_ids, resolwe)
+
+    inputs = {
+        'reads': reads,
+        'expressions': expressions,
+        'name': name or auto_name,
+    }
+    geo = resolwe.get_or_run(slug='prepare-geo-rnaseq', input=inputs)
+
+    if collection:
+        collection.add_data(geo)
+
+    return geo
+
+
+def prepare_geo(resource, name=None, *types):
+    """Run several prepare geo functions on a resource.
+
+    :param list types: list of sequencing types of the samples in the
+    resource. If none are given, the function is run on all types.
+    Options are: ChIP-Seq, RNA-Seq
+    :param str name: name of the prepare GEO tarball and table
+
+    """
+
+    type_to_function = {
+        'rna-seq': prepare_geo_rnaseq,
+        'chip-seq': prepare_geo_chipseq,
+    }
+
+    results = []
+
+    if not types:
+        types = type_to_function.keys()
+
+    for seq_type in types:
```

Comment:
Add a check that ``types`` are valid. Something like:

    invalid_types = [type_ for type_ in types if type_ not in type_to_function.keys()]
    if invalid_types:
        raise ValueError(
            "Invalid types '{}', the following are supported: '{}'".format(
                "', '".join(invalid_types),
                "', '".join(type_to_function.keys()),
            )
        )